docs/html/development/architecture/index.rst

```diff
@@ -8,13 +8,215 @@ Architecture of pip's internals
     interested in helping out, please let us know in the `tracking issue`_.
 
 .. note::
-    Direct use of pip's internals is *not supported*.
-    For more details, see :ref:`Using pip from your program`.
+    Direct use of pip's internals is *not supported*, and these internals
+    can change at any time. For more details, see :ref:`Using pip from
+    your program`.
 
 
 .. toctree::
     :maxdepth: 2
 
 
+Finding and choosing files (``index.py`` and ``PackageFinder``)
```

Comment:
Could you move this to a separate file named `index-interactions.rst` and add `index-interactions` (or a different appropriate name) to the toctree above?